domain/machine/state/state_test.go

```diff
@@ -928,3 +932,145 @@ func (s *stateSuite) TestAppliedLXDProfileNamesNoErrorEmpty(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Check(profiles, gc.HasLen, 0)
 }
+
+// TestSetRunningAgentBinaryVersionSuccess asserts that if we attempt to set the
+// running agent binary version for a machine that doesn't exist we get back
+// an error that satisfies [machineerrors.MachineNotFound].
+func (s *stateSuite) TestSetRunningAgentBinaryVersionMachineNotFound(c *gc.C) {
+	machineUUID, err := uuid.NewUUID()
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = s.state.SetRunningAgentBinaryVersion(
+		context.Background(),
+		machineUUID.String(),
+		coreagentbinary.Version{
+			Number: jujuversion.Current,
+			Arch:   corearch.ARM64,
+		},
+	)
+	c.Check(err, jc.ErrorIs, machineerrors.MachineNotFound)
+}
+
+// TestSetRunningAgentBinaryVersionNotSupportedArch tests that if we provide an
+// architecture that isn't support by the database we get back an error
+// satisfying [coreerrors.NotValid].
```

Comment:
just a suggestion here, sometime we use `an error satisfying` and sometimes we use `an error that satisfies`, it would be good to keep one. LOL